AzureIoTHub.agent.lib.nut

```diff
@@ -629,513 +604,842 @@ class AzureIoTHub {
+        //     onDisconnect : Function      Callback called every time the device is disconnected
+        //          (optional)              The callback signature:
+        //                                  onDisconnect(error), where
+        //                                      error : Integer     0 if the disconnection was caused by the disconnect() method,
+        //                                                          an error code which explains a reason of the disconnection otherwise.
+        //     options : Table              Key-value table with optional settings.
+        //          (optional)
+        //
+        // Returns:                         AzureIoTHub.Client instance created.
+        constructor(deviceConnStr, onConnect, onDisconnect = null, options = {}) {
+            _state = S_DISCONNECTED;
+            _options = DEFAULT_CLIENT_OPTIONS;
+
+            _onConnectCb      = onConnect;
+            _onDisconnectCb   = onDisconnect;
+
+            _connStrParsed = AzureIoTHub.ConnectionString.Parse(deviceConnStr);
+            _mqttclient = mqtt.createclient();
+            _mqttclient.onconnect(_onConnect.bindenv(this));
+            _mqttclient.onconnectionlost(_onDisconnect.bindenv(this));
+            _mqttclient.onmessage(_onMessage.bindenv(this));
+
+            foreach (optName, optVal in options) {
+                _options[optName] <- optVal;
             }
-        }
 
-        function disconnect() {
-            _connection.close();
-            _resetConnectionTables();
+            _fillTopics(_connStrParsed.DeviceId);
         }
 
-        // done cb params - err
-        function sendEvent(message, done = null) {
-            if ( !_isOpen(_sessions.event) ) {
-                if (done) done("Cannot send while disconnected.");
-            } else {
-                if ( !_isOpen(_senders.event) ) {
-                    _log("creating event sender");
-                    // add message to queue
-                    _msgQueue.push({"msg" : message, "cb" : done});
-                    // create a new sender & send message
-                    _openEventSender();
-                } else {
-                    _log("event sender open")
-                    _sendEvent(message, done);
-                }
-            }
-        }
-
-        // done is called everytime a message is available, params - error, data
-        function receive(done) {
-            _handlers.onEvent <- done;
+        // Opens a connection to Azure IoT Hub.
+        //
+        // Returns:                         Nothing.
+        function connect() {
+            if (_state == S_DISCONNECTED) {
+                _log("Connecting...");
 
-            if (done == null) {
-                _receivers.event.close();
-                // clear out handlers
-                return this;
-            }
+                local devPath = "/" + _connStrParsed.DeviceId;
```

Comment:
Shouldn't it be `deviceId`, lowercase?

Reply:
`DeviceId` is a field of the table created after parsing a string containing `HostName=xxx;DeviceId=xxx;SharedAccessKey=xxx`